src/gluonts/model/prophet/_predictor.py

```diff
@@ -1,54 +1,147 @@
 # Standard library imports
-from typing import Callable, Dict, Iterator, Optional
+from typing import Callable, Dict, Iterator, NamedTuple, Optional, List
 
 # Third-party imports
+import numpy as np
 import pandas as pd
 
 # First-party imports
 from gluonts.core.component import validated
-from gluonts.dataset.common import Dataset
-from gluonts.model.forecast import SampleForecast
+from gluonts.dataset.common import DataEntry, Dataset
+from gluonts.model.forecast import SampleForecast, Config
 from gluonts.model.predictor import RepresentablePredictor
 
+try:
+    from fbprophet import Prophet
+except ImportError as e:
+    pass
+
+PROPHET_IS_INSTALLED = 'Prophet' in locals()
+
 USAGE_MESSAGE = """
-The `ProphetPredictor` is a thin wrapper for calling the Prophet package.
-In order to use it you need to install fbprophet
+Cannot import `fbprophet`.
 
-pip install fbprophet
+The `ProphetPredictor` is a thin wrapper for calling the `fbprophet` package.
+In order to use it you need to install it using one of the following two
+methods:
 
+    # 1) install fbprophet directly
+    pip install fbprophet
+
+    # 2) install gluonts with the Prophet extras
+    pip install gluonts[Prophet]
 """
 
 
+def feat_name(i: int) -> str:
+    """The canonical name of a feature with index `i`."""
+    return f'feat_dynamic_real_{i:03d}'
+
+
+class ProphetDataEntry(NamedTuple):
+    """
+    A named tuple containing relevant base and derived data that is
+    required in order to call Prophet.
+    """
+
+    train_length: int
+    prediction_length: int
+    start: pd.Timestamp
+    target: np.ndarray
+    feat_dynamic_real: List[np.ndarray]
+
+    @property
+    def train_input(self) -> pd.DataFrame:
```

Comment:
Why is this called `train_input`? This seems to be converting the data fields in the object into a `pd.DataFrame`, maybe `to_dataframe` is a better name?

Reply:
Because it is passed to the `prophet.fit` method. I renamed the property to `prophet_training_data`.